### PR TITLE
`DoesIntersect` - add intersection between point and element, point and solid, and solid vs solid

### LINKED ISSUE #1421: DoesIntersect - intersection between point and element
#### Description:

Add method to check if point intersects with element.

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #1421

Add `DoesIntersect` method to check the intersection between point and element.

### CODE REVIEW

File: Revit_Core_Engine/Query/DoesIntersect.cs [hunk 1]
```diff
@@ -109,6 +110,80 @@ public static bool DoesIntersect(this BoundingBoxXYZ bbox, Element element)
...
+        [Output("bool", "Result of the intersect checking.")]
+        public static bool DoesIntersect(this Solid solid1, Solid solid2)
+        {
+            if (solid1 == null || solid2 == null)
+            {
+                return false;
+            }
+
+            Solid intersectSolid = BooleanOperationsUtils.ExecuteBooleanOperation(solid1, solid2, BooleanOperationsType.Intersect);
+            return Math.Abs(intersectSolid.Volume) > BH.oM.Geometry.Tolerance.Distance;
```
Comment (vietle-bh) [inline Revit_Core_Engine/Query/DoesIntersect.cs:184]:
```suggestion
            return Math.Abs(intersectSolid.Volume) > Math.Pow(Tolerance.Distance, 3);
```

**Review by vietle-bh — COMMENTED** [review]
Everything else looks great 👍

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check dataset-compliance
@BHoMBot check ready-to-merge

**Pull request merged**